nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -535,6 +565,46 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   return transformed_plan;
 }
 
+inline double RegulatedPurePursuitController::dot(double u_x, double u_y, double v_x, double v_y)
+{
+  // Supports only 2D
+  return  (u_x * v_x) + (u_y * v_y);
+}
+
+double RegulatedPurePursuitController::detectPathOrientation(
+  const geometry_msgs::msg::PoseStamped & pose )
+{
+  double path_direction_changed = 0.0;
+  double cusp_distance = 0.0;
+
+  // Iterating through the global path to determine the position of the cusp
+  for (auto iter = 1; iter <= (int)global_plan_.poses.size(); ++iter) {
```

Comment:
This is not an iterator, please don't mix those up. This should be `i` or `pose_id` or something.

Reply:
And you can use `unsigned int` rather than `auto` so that you don't need the implicit cast from `size()` (so remove that part)

Reply:
Make sure to remove `auto` and replace with `unsigned int`. then you can remove the `(int)`

Reply:
oops! sorry!